packages/backend/jobs-manager/service/src/modules/database/views/views.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model, Types } from 'mongoose';
+import { View, ViewDocument } from './views.model';
+
+@Injectable()
+export class ViewService {
+  private logger = new Logger(ViewService.name);
+
+  constructor(
+    @InjectModel('view')
+    private readonly viewModel: Model<View>,
+  ) {}
+
+  public async getAll(): Promise<
+    Pick<ViewDocument, 'name' | 'icon' | 'isPinned' | '_id'>[]
+  > {
+    return await this.viewModel.find({}, '_id name icon isPinned');
+  }
+
+  public async getView(id: string) {
+    return await this.viewModel.findById(id);
+  }
+
+  public async createView(view: View) {
+    return await this.viewModel.create(view);
+  }
+
+  public async updateView(id: string, view: View) {
```

Comment:
On devrait omit l'id 

```suggestion
  public async updateView(id: string, view: Omit<View, '_id' | 'id>) {
```

Reply:
Normalement, `View` n'a pas de id ou _id, et c'est `ViewDocument` qui a cette information